core/dbt/parser/manifest.py

```diff
@@ -214,15 +212,52 @@ def load(self):
         if self.saved_manifest is not None:
             partial_parsing = PartialParsing(self.saved_manifest, self.manifest.files)
             skip_parsing = partial_parsing.skip_parsing()
-            if not skip_parsing:
+            if skip_parsing:
+                # nothing changed, so we don't need to generate project_parser_files
+                self.manifest = self.saved_manifest
+            else:
                 # create child_map and parent_map
                 self.saved_manifest.build_parent_and_child_maps()
                 # files are different, we need to create a new set of
                 # project_parser_files.
-                project_parser_files = partial_parsing.get_parsing_files()
-                self.partially_parsing = True
-
-            self.manifest = self.saved_manifest
+                try:
+                    project_parser_files = partial_parsing.get_parsing_files()
+                    self.partially_parsing = True
+                    self.manifest = self.saved_manifest
+                except Exception:
+                    # pp_files should still be the full set and manifest is new manifest,
+                    # since get_parsing_files failed
+                    logger.info(
+                        "Partial parsing enabled but an error occurred. Full parsing resumed."
+                    )
+
+                    # Get traceback info
+                    tb_info = traceback.format_exc(10)
+                    formatted_lines = tb_info.splitlines()
+                    (_, line, method) = formatted_lines[-3].split(', ')
+                    exc_info = {
+                        "traceback": tb_info,
+                        "exception": formatted_lines[-1],
+                        "code": formatted_lines[-2],
+                        "location": f"{line} {method}",
```

Comment:
I didn't realize these fields were all being munged out of the full traceback; makes sense. Thanks for doing the light cleanup! It will be a lot easier to identify common failure modes. Also glad we're still sending the raw traceback, to enable debugging.

There are some absolute file paths in the traceback, not to resources in someone's dbt project, but in the sense of where someone has installed dbt / python. I don't think that's avoidable, though, and it's hardly identifying—it's probably the path determined by a package manager (Homebrew etc).

Reply:
Yes, I figured we could sort errors out better with individual fields.